jerry-core/ecma/operations/ecma-jobqueue.h

```diff
@@ -26,23 +26,26 @@
  */
 
 /**
- * Jerry job handler function type
+ * Job queue item types.
  */
-typedef ecma_value_t (*ecma_job_handler_t) (void *job_p);
+typedef enum
+{
+  ECMA_JOB_PROMISE_REACTION, /**< promise reaction job */
+  ECMA_JOB_PROMISE_THENABLE, /**< promise thenable job */
+} ecma_job_queue_item_type_t;
 
 /**
  * Description of the job queue item.
  */
-typedef struct ecma_job_queueitem_t
+typedef struct ecma_job_queue_item_t
 {
-  struct ecma_job_queueitem_t *next_p; /**< points to next item */
-  ecma_job_handler_t handler; /**< the handler for the job*/
-  void *job_p; /**< points to the job */
-} ecma_job_queueitem_t;
+  struct ecma_job_queue_item_t *next_p; /**< points to next item */
```

Comment:
Is there any plan to support more than 8 types? If it isn't the type could be stored in the lowest 3 bit of the pointer. It'd slightly make the usage harder but we could save memory with it.